### DISCUSSION TITLE
What is the model-best output based on?

### DISCUSSION THREAD

**Discussion by erotavlas** [explosion/spaCy#8839] [Help: Other Questions] [answered]
How is the output in model-best folder determined?  What makes it best compared to the  model-last output from the final training iteration.

**Comment by polm** [comment, marked as answer]
When training, there's period output as the trainer runs an evaluation step on dev data. That output includes a "score", which is used to select the best model.

**Reply by erotavlas** [reply]
in order to create my final model I use all my data (train + dev) which is evaluated on a final test set.  
therefore I set both train and dev in the config to (train + dev) - so they are the same.  

based on what you are saying, should I swap my dev set for my final test set when I'm done hyperparameter tuning and validation phase (i.e. when ready to train the final model for production use)?

**Reply by polm** [reply]
Yes, that would be better. If your train and dev set are the same you can overfit without realizing it.
